### ISSUE TITLE
NTI gives incorrect error:  "property reference is typed as unknown”

### ISSUE THREAD

**Issue by myphysicslab** [google/closure-compiler#2442]
In a constructor I define a boolean field called `debugCF`:

```javascript
/**
* @param {string} name for debugging, this distinguishes whether this is used for
*     contact forces or collision impulses
* @param {!myphysicslab.lab.util.Random} pRNG  pseudo random number generator, used to
*    randomly decide order in which to calculate forces
* @param {number=} tolerance used to decide when numbers are equal or zero;
*    default is 1E-10
* @constructor
* @final
* @struct
*/
myphysicslab.lab.engine2D.ComputeForces = function(name, pRNG, tolerance) {
  /** debug compute forces
  * @type {boolean}
  * @private
  */
  this.debugCF = false;
```
later on in the constructor I get the conformance rule warning `BanUnknownTypedClassPropsReferences` but only when using NTI (new type inference)

```javascript
  /** The Next Contact Policy to use for deciding order in which to treat contacts.
  * @type {number}
  * @private
  */
  this.nextContactPolicy = ComputeForces.NEXT_CONTACT_HYBRID;
  if (goog.DEBUG && this.debugCF)
    this.print('nextContactPolicy=' + this.nextContactPolicy);
```
here is the warning:
```
src/lab/engine2D/ComputeForces.js:359: WARNING - Violation: property reference is typed as “unknown”
The property "debugCF" on type "myphysicslab.lab.engine2D.ComputeForces{DEFER_SINGULAR:boolean, WARNINGS:boolean, debugCF:boolean}"
  if (goog.DEBUG && this.debugCF)
                    ^^^^^^^^^^^^
```
Here is a link to this line in the source code:
https://github.com/myphysicslab/myphysicslab/blob/master/src/lab/engine2D/ComputeForces.js#L359

To reproduce:  

1. The readme at https://github.com/myphysicslab/myphysicslab tells how to build the project (not difficult especially if you already have closure-compiler and closure-library).  Don't do the "make" step yet.

2. uncomment those lines shown in the above link

3. Don't need to make everything, instead use this command:
```
make chain LOCALE=en
```

This might be similar to #2426.

**Comment by dimvar** [comment]
I have an upcoming fix for #2426. If this issue is still happening after that fix, reply here and I'll take a look.

**Comment by dimvar** [comment]
@myphysicslab #2426 is now fixed. Can you check if this issue still happens?

**Comment by myphysicslab** [comment]
Will do as soon as I can.  I'm traveling over next couple days.

**Comment by myphysicslab** [comment]
Fixed.  That warning no longer happens after updating compiler to b85c29855a714cae446f61c8a7b2ccaac747722b.
